packages/core/src/model.types.ts

```diff
@@ -38,9 +40,24 @@ export interface Model<
   actions: Prop<TModelCreators, 'actions'>;
   reset: () => AssignAction<TContext, any>;
   createMachine: (
-    config: MachineConfig<TContext, any, TEvent, TAction>,
-    implementations?: Partial<MachineOptions<TContext, TEvent, TAction>>
-  ) => StateMachine<TContext, any, TEvent>;
+    config: MachineConfig<TContext, any, TEvent, TExtra, TAction>,
+    implementations?: Partial<MachineOptions<TContext, TEvent, TExtra, TAction>>
+  ) => StateMachine<
+    TContext,
+    any,
+    TEvent,
+    { value: any; context: TContext },
+    TExtra
+  >;
+  withInput: <TInput extends {}>(
```

Comment:
Most likely you've wanted a different constraint here:
```suggestion
  withInput: <TInput extends Record<string, unknown>>(
```

Unfortunately, `{}` when used as a constraint is not really constraining the generic to be a subtype of an empty object. It's some kind of "allow all" (well, probably not exactly that but this always confuses me and I've stopped trying to understand it). Check out [the demo on TS playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABFApgZygHgCqJQD1TABM1EBvAXwD4AKASgC5FdyAoRRAJxShC6RgQAG2GIAhmXFgAngG42lNm1QZMULiBR16C1VgBEBnXvRYAjACYAzCZVnMVO-seJgcOMwMAjcVwOINAwKbKCQsAjIZpY4eIQoJGQASigQcFzEmBhcMGAA5gA0iOAA1mBwAO5g1DrMrBzcvPyCImKSErIKSvYYMRpaztGYRoO9mFa2wT1QMU5T+rNuHl6+-oF2QA)